pkg/daemon/controller/controller.go

```diff
@@ -488,9 +488,21 @@ func (sc *ServiceController) resolveNodeVariables(svc servicescm.Service) (map[s
 func (sc *ServiceController) resolvePowershellVariables(svc servicescm.Service) (map[string]string, error) {
 	vars := make(map[string]string)
 	for _, script := range svc.PowershellPreScripts {
+		// replaces the values in the PowershellPreScripts path with the node args
+		if len(script.NodeArgs) != 0 {
+			psNodeVars, err := sc.resolveNodeVariables(script.NodeArgs)
+			if err != nil {
+				return nil, err
+			}
+			pwshCommand := script.Path
+			for key, value := range psNodeVars {
+				pwshCommand = strings.ReplaceAll(pwshCommand, key, value)
+			}
+			script.Path = pwshCommand
+		}
```

Comment:
consider breaking this out into a helper function to ease unit testing

Reply:
Do you think we should break it out and test the individual function, instead of testing within ResolveNodeVariables?